python/ray/data/dataset.py

```diff
@@ -2395,6 +2398,151 @@ def iter_batches(
 
         stats.iter_total_s.add(time.perf_counter() - time_start)
 
+    def iter_torch_batches(
+        self,
+        *,
+        prefetch_blocks: int = 0,
+        batch_size: Optional[int] = None,
+        dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
+        device: Optional[str] = None,
+        drop_last: bool = False,
+        local_shuffle_buffer_size: Optional[int] = None,
+        local_shuffle_seed: Optional[int] = None,
+    ) -> Iterator[TorchTensorBatchType]:
+        """Return a local batched iterator of Torch Tensors over the dataset.
+
+        This iterator will yield single-tensor batches if the underlying dataset
+        consists of a single column; otherwise, it will yield a dictionary of
+        column-tensors. If looking for more flexibility in the tensor conversion (e.g.
```

Comment:
Please remove the mention of to_torch (we should also mark it as `@Deprecated`), here and in other docs. Here are mentions we should change:

```
source/ray-air/check-ingest.rst
43:typically calls ``iter_batches``, ``to_tf``, or ``to_torch`` to iterate over the dataset reader retrieved by ``get_dataset_shard``.

source/ray-air/examples/torch_incremental_learning.ipynb
544:    "    dataset_shard = session.get_dataset_shard(\"train\").to_torch(\n",

source/ray-air/examples/torch_image_example.ipynb
145:    "Next, let's represent our data using pandas dataframes instead of tuples. This lets us call methods like {py:meth}`Dataset.to_torch <ray.data.Dataset.to_torch>` later in the tutorial."
283:    "* We call {py:func}`session.get_dataset_shard <ray.air.session.get_dataset_shard>` and {py:meth}`Dataset.to_torch <ray.data.Dataset.to_torch>` to convert a subset of our training data to a Torch dataset.\n",
305:    "    train_dataset_shard: torch.utils.data.Dataset = session.get_dataset_shard(\"train\").to_torch(\n",

source/train/user_guide.rst
454:        train_torch_dataset = train_data_shard.to_torch(
459:        validation_torch_dataset = validation_data_shard.to_torch(
1203:            .to_torch(batch_size=config["batch_size"])

source/data/accessing-datasets.rst
66:For ingestion into one or more Torch trainers, Datasets offers a :meth:`ds.to_torch()
67:<ray.data.Dataset.to_torch>` API that returns a
78:  operations in conjunction with :meth:`ds.to_torch() <ray.data.Dataset.to_torch>`
89:that we may want to split into separate tensors. By informing ``ds.to_torch()`` of the
110:each tensor). See the :meth:`.to_torch() API docs <ray.data.Dataset.to_torch>` for

source/data/pipelining-compute.rst
77:   Alternatively, you may consider local shuffle after converting to_tf() or to_torch(), if simple shuffle is sufficient.

source/data/dataset-tensor-support.rst
189::meth:`ds.to_torch() <ray.data.Dataset.to_torch>` and
219:    torch_ds = ds.to_torch(
269:``to_torch()`` a ``feature_columns=[["feature_1"], ["feature_2"]]`` argument in order to
271:``to_torch()``, if isolating single columns as in the ``"feature_1"`` + ``"feature_2"``
295:    torch_ds = ds.to_torch(

source/data/key-concepts.rst
51:A DatasetPipeline is an unified iterator over a (potentially infinite) sequence of Ray Datasets, each of which represents a *window* over the original data. Conceptually it is similar to a `Spark DStream <https://spark.apache.org/docs/latest/streaming-programming-guide.html#discretized-streams-dstreams>`__, but manages execution over a bounded amount of source data instead of an unbounded stream. Ray computes each dataset window on-demand and stitches their output together into a single logical data iterator. DatasetPipeline implements most of the same transformation and output methods as Datasets (e.g., map, filter, split, iter_rows, to_torch, etc.).

source/data/advanced-pipelines.rst
91:        for batch in pipe.to_torch():

source/data/dataset.rst
287:     - :meth:`ds.to_torch() <ray.data.Dataset.to_torch>`

source/data/memory-management.rst
71::meth:`ds.to_torch() <ray.data.Dataset.to_torch>`, etc.) or if

source/data/doc_code/key_concepts.py
75:    .to_torch()

source/data/doc_code/accessing_datasets.py
95:torch_ds: torch.utils.data.IterableDataset = ds.to_torch(batch_size=2)
125:torch_ds: torch.utils.data.IterableDataset = ds.to_torch(
221:        for batch in shard.to_torch(batch_size=256):

source/ray-core/_examples/datasets_train/datasets_train.py
290:   .to_torch_dataset()
418:    test_torch_dataset = test_dataset.to_torch(
440:        train_torch_dataset = train_dataset.to_torch(
```